docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -15,143 +14,286 @@ keywords:
+    LowerLimit   float64 `json:"lowerLimit"`
+    TickInterval float64 `json:"tickInterval"`
+}
+```
+
+In this example, have are creating a ticker and an infinity loop based on this ticker. Whenever the ticker times out, we will enter the second case of the `select` and generate a random number `randomValue`. After we will create a data frame using `data.NewFrame`, that contains the `randomValue` and the current time, and send it using `sender.SendFrame`. This loop will run indefinitely until we close the channel.
+
+:::tip
+
+You can remove the method `QueryData` that is created by the scaffold and all the code related to it since we are not implementing the backend.`QueryDataHandler` anymore.
+
+:::
+
+## Step 4: Build the plugin
+
+The coding part of the plugin development process is done. Now we need to generate our plugin package to be able to run it in Grafana. Since our plugin has both a frontend and a backend part, we do this in two steps.
+
+1. Build the frontend:
+
+```
+npm install
+npm run build
+```
+
+This should download all the dependencies and create the frontend plugin files in the `./dist` directory.
+
+1. Compile the backend code and generate the plugin binaries. For that you should have `mage` installed and then you simply need to run:
+
+```
+mage -v
+```
+
+This command compiles the Go code and generate binaries for all Grafana-supported platforms in `./dist`.
+
+## Step 5: Run the plugin
+
+Once the plugin is built, you can copy `./dist` together with its content, rename it to the `id` of your plugin, put it in the plugin's path of your Grafana instance, and test it.
+
+However, there is an easier way of doing all this.
+
+1. Simply run:
+
+`npm run server`
+
+This command runs a Grafana container using docker-compose and puts the built plugins in the right place.
+
+1. To verify the build, go to Grafana at `https://localhost:3000`.
+
+:::note
+
+You may be using Docker Compose instead of docker-compose. If that is your case, you just need to change the command in the `package.json` in the project's root directory.
+
+:::
+
+## Step 6: Test the plugin
+
+1. From Grafana at `https://localhost:3000`, use the default credentials: username `admin` and password `admin`. If you're not presented with a login page, click **Sign in** in the top of the page and insert the credentials.
+
+1. Add the data source. Since we are running in our `docker-compose` environment, we don't need to install it, and it will be directly available for usage. Go to **Connections > Data sources**, using the left menu as shown in the following image:
+
+1. A new page opens, and then click **Add data source**. Grafana opens another page where you can search for the data source name that we've just created.
+
+1. Click on the data source's card, and then click **Save & test** on the next page.
+
+![Data sources - Grafana UI.](/img/data-sources-menu.png)
+
+1. If a randomized error occurs, just click **Save & test** until a success message appears:
```

Comment:
What is this about?

Reply:
This comes from the code provided by the scaffold created by plugin tools
https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/backend/pkg/plugin/datasource.go#L96-L109

But maybe we can add an extra step in the tutorial to change it to

```go
func (d *Datasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
	return &backend.CheckHealthResult{
		Status:  backend.HealthStatusOk,
		Message: "Data source is working",
	}, nil
}
```
What do you think about that @josmperez ?
Or you can make it as a tip as you did with the `QueryData`

Reply:
@oshirohugo - I have added this code, but you don't state where the code should be added. Can you check the change and suggest the point of addition?

Reply:
I think it's better to add this part of the code to the end of `Step 3: Set up the plugin backend`, since it should also be included also in `pkg/plugin/datasource.go`.

We can say something like "to prevent a randomize error, change the `CheckHealth` method provided by default for the following code..."

We should also state that this part has nothing to do with the streaming plugin itself. It's just to avoid the error that will be present by method provided by default.

Reply:
sorry, I misunderstood earlier. I've moved the new step to the bottom (#7) of step 3.